cmd/zpool/zpool_iter.c

```diff
@@ -374,6 +374,18 @@ for_each_vdev_run_cb(zpool_handle_t *zhp, nvlist_t *nv, void *cb_vcdl)
 		}
 	}
 
+	/* Check for whitelisted vdevs here, if any */
+	for (i = 0; i < vcdl->vdev_names_count; i++) {
+		if (strcmp(vcdl->vdev_names[i], zpool_vdev_name(g_zfs, zhp, nv,
+		    vcdl->cb_name_flags)) == 0)
+			break; /* match */
+	}
+
+	/* If we whitelisted vdevs, and this isn't one of them, then bail out */
+	if (vcdl->vdev_names_count && (i == vcdl->vdev_names_count)) {
+		return (0);
+	}
```

Comment:
[style] No { } around single line conditionals.  This works but how about changing it to something a little more self documenting like this (untested):

```
        int match = 0, i;
        ...

        /* Check for whitelisted vdevs here if provided */
        if (vcdl->vdev_names_count && vcdl->vdev_names != NULL) {
                for (i = 0; i < vcdl->vdev_names_count; i++) {
                        if (strcmp(vcdl->vdev_names[i], zpool_vdev_name(
                            g_zfs, zhp, nv, vcdl->cb_name_flags)) == 0) {
                                match = 1;
                                break; 
                        }
                }

                if (!match)
                        return (0);
        }
```